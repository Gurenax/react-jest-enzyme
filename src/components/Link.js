import React from 'react'

const Link = ({
  title,
  url,
  onLinkClick
}) => (
  <div className='output-area' onClick={onLinkClick}>
    <h2>A link to {title}</h2>
    <a href={url} onClick={()=>{ alert('Clicked!') }}>
      {title}
    </a>
  </div>
)

export default Link
