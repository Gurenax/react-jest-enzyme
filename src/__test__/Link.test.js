import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Link from '../components/Link'

describe('Link', () => {
  /*
  *   Test by counting number of elements
  */
  it('renders a link', () => {
    const output = shallow(<Link />)
    expect(output.find('a').length).toEqual(1)
  })

  it('renders an output area', () => {
    const output = shallow(<Link />)
    expect(output.find('div.output-area').length).toEqual(1)
  })

  it('renders a heading inside the output area', () => {
    const output = shallow(<Link />)
    expect(output.find('div.output-area').find('h2').length).toEqual(1)
  })

  it('renders the link inside the output area', () => {
    const output = shallow(<Link />)
    expect(output.find('div.output-area').find('a').length).toEqual(1)
  })

  /*
  *   Test by exact html
  */
  it('renders the Link output area', () => {
    const output = shallow(<Link title='Google' url='http://www.google.com'/>)
    const expected =
      '<div class="output-area">' +
      '<h2>A link to Google</h2>' +
      '<a href="http://www.google.com">'+
      'Google' +
      '</a>' +
      '</div>'
    const result = output.find('div.output-area').html()
    expect(result.indexOf(expected) > -1).toEqual(true)
  })

  /*
  * Test by matching a snapshot
  */
  it('renders correctly', () => {
    const output = shallow(
      <Link title="mockTitle" url="mockUrl" />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  /*
  * Test events
  */
  it('handles the click event and calls a window.alert', () => {
    window.alert = jest.fn()
    const output = shallow(
      <Link title="mockTitle" url="mockUrl" />
    )
    
    output.find('a').simulate('click')
    expect(window.alert).toHaveBeenCalledWith('Clicked!')
  })

  /*
  * Test events when it's a custom function
  */
  it('calls a function when clicked', () => {
    const doSomethingCool = jest.fn() // Mock function doSomethingCool()
    const output = shallow(
      <Link title="mockTitle" url="mockUrl" onLinkClick={() => {
        doSomethingCool()
      }} />
    )
    output.find('div').simulate('click')
    expect(doSomethingCool).toHaveBeenCalledTimes(1)
  })

})