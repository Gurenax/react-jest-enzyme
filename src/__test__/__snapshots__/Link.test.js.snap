// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="output-area"
>
  <h2>
    A link to 
    mockTitle
  </h2>
  <a
    href="mockUrl"
    onClick={[Function]}
  >
    mockTitle
  </a>
</div>
`;
